src/components/app-login/app-login.tsx

```diff
@@ -0,0 +1,59 @@
+import { Component, h, Method, Prop, State } from '@stencil/core';
+
+@Component({
+  tag: 'app-login',
+  styleUrl: 'app-login.css',
+  shadow: true,
+})
+
+
+
+export class AppLogin {
+
+  @Prop() loading: boolean = false;
+
+  @State() data: any = {};
+
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.data[name] = value;
```

Comment:
this type of update not recommended as stencil will not know if state is updated so it can rerender
see "complex types" case in stencil https://stenciljs.com/docs/state
luckily its not giving issue for this case, we can discuss this tomorrow in detail